activerecord/lib/active_record/relation/predicate_builder.rb

```diff
@@ -95,6 +95,11 @@ def create_binds_for_hash(attributes)
             next
           when value.is_a?(Relation)
             binds += value.bound_attributes
+          when value.is_a?(Array) && !table.type(column_name).respond_to?(:subtype)
```

Comment:
If removed the type condition, `test_where_by_attribute_with_array` fails.
https://github.com/rails/rails/blob/master/activerecord/test/cases/adapters/postgresql/array_test.rb#L293-L297

```ruby
  def test_where_by_attribute_with_array
    tags = ["black", "blue"]
    record = PgArray.create!(tags: tags)
    assert_equal record, PgArray.where(tags: tags).take
  end
```

```
# correct
SELECT  "pg_arrays".* FROM "pg_arrays" WHERE "pg_arrays"."tags" = $1 LIMIT $2  [["tags", "{black,blue}"], ["LIMIT", 1]]

# removed type condition
SELECT  "pg_arrays".* FROM "pg_arrays" WHERE "pg_arrays"."tags" IN ('black', 'blue') LIMIT $1  [["LIMIT", 1]]
```